src/webui/api/torrentscontroller.cpp

```diff
@@ -481,7 +481,6 @@ void TorrentsController::addAction()
     }
 
     BitTorrent::AddTorrentParams params;
-    // TODO: Check if destination actually exists
```

Comment:
When the user specifies "Skip checking", it means that qBittorrent will not check the files, but will begin to seed them, assuming they already exist. This comment means "it would be nice to check the files existence before".
Your changes is meaningless since qBittorrent creates save folder when add torrent to session.

Reply:
In the WebUI the user can specify whatever path they'd like. There's no guarantee that the path will actually exist- and that it isn't complete gibberish- so that's what this check is for. This has nothing to do with the "Skip checking" option.

Reply:
>This has nothing to do with the "Skip checking" option.

Why did you delete this comment? It's just about "Skip checking" option.

>There's no guarantee that the path will actually exist- and that it isn't complete gibberish- so that's what this check is for.

Then you should try to create it, but not check for its existence. As I said, it's allowed case when path doesn't exist before we add torrent.

Reply:
> Then you should try to create it, but not check for its existence. As I said, it's allowed case when path doesn't exist before we add torrent.

I think you're describing the case where the last directory doesn't exist. I'm focusing on the issue where the user specifies a path `/mnt1/Disk/Torrents/Downloads` where `/mnt1` doesn't exist. Surely in this case we shouldn't create that entire path, but throw an error.

Reply:
I don't want this commit to be the focus of this PR. If it's that contentious a change I'll gladly remove it.

Reply:
>I don't want this commit to be the focus of this PR.

Me too. I just wanted to clear the space for the main review.
>If it's that contentious a change I'll gladly remove it.

It is not as unambiguous as it may seem at first glance. It is better to move it in a separate PR.
>I think you're describing the case where the last directory doesn't exist.

No, I'm describing common case. It should create save path even if it multilevel.

Reply:
> It is not as unambiguous as it may seem at first glance. It is better to move it in a separate PR.

I'll move it to a separate PR.

> No, I'm describing common case. It should create save path even if it multilevel.

In the case I described (`/mnt1/Disk/Torrents/Downloads`) would you expect qBittorrent to create all directories, or would you expect an error for `/mnt1` not existing? I personally would expect the latter.

Reply:
I didn't say what I expected. I was talking about how it works now.
But, IMO, this behavior is quite expected, otherwise users (especially, remote users) will get a hell of a headache (it breaks some automations).

P.S. Let's continue this discussion in its own topic.